### PR TITLE
Feature le

### PR DESCRIPTION
**Pull request by Leho123**

### CODE REVIEW

File: controllers/Cart/Cart_controller.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+$dsn = "mysql:host=$hostname;dbname=$database;charset=utf8mb4";
+
+
+try {
+    $connection = new PDO($dsn, $username, $password);
+    $connection->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+} catch (PDOException $e) {
+    die("Connection failed: " . $e->getMessage());
+}
```
Comment (namdang29) [inline controllers/Cart/Cart_controller.php]: All of this is already in the config, no need to rewrite it

File: controllers/Cart/Cart_controller.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+
+if ($_SERVER["REQUEST_METHOD"] == "POST") {
+
```
Comment (namdang29) [inline controllers/Cart/Cart_controller.php]: Only one line between each block

File: controllers/Cart/Cart_controller.php [hunk 3]
```diff
@@ -0,0 +1,90 @@
+<?php
+session_start();
+
```
Comment (namdang29) [inline controllers/Cart/Cart_controller.php]: Only one line between each block

File: controllers/Cart/Cart_controller.php [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+       
+        unset($_SESSION['cart']);
+    }
+}
+?>
+
+
+
```
Comment (namdang29) [inline controllers/Cart/Cart_controller.php]: remove

File: controllers/Checkout/Checkout.controller.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+$password = "your_password";
+$dbname = "database_project_php";
+
+$conn = new mysqli($servername, $username, $password, $dbname);
+
+
+if ($conn->connect_error) {
+    die("Kết nối đến cơ sở dữ liệu thất bại: " . $conn->connect_error);
+}
+
```
Comment (namdang29) [inline controllers/Checkout/Checkout.controller.php]: All of this is already in the config, no need to rewrite it

File: controllers/Cart/Cart_controller.php [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        $zipCode = $_POST['zipCode'];
+        $phoneNo = $_POST['phoneNo'];
+       
+       
+        $stmt = $connection->prepare("INSERT INTO orders (userId, address, zipCode, phoneNo, amount, paymentMode, orderStatus) VALUES (?, ?, ?, ?, ?, ?, '0')");
+        $stmt->execute([$userId, $address, $zipCode, $phoneNo, $totalAmount, $paymentMode]);
+       
+       
+        $orderId = $connection->lastInsertId();
+
```
Comment (namdang29) [inline controllers/Cart/Cart_controller.php]: Handle in model

File: controllers/Cart/Cart_controller.php [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+       
+        foreach ($_SESSION['cart'] as $pizzaId => $item) {
+            $quantity = $item['quantity'];
+
+
+            $stmt = $connection->prepare("INSERT INTO orderItems (orderId, pizzaId, itemQuantity) VALUES (?, ?, ?)");
+            $stmt->execute([$orderId, $pizzaId, $quantity]);
```
Comment (namdang29) [inline controllers/Cart/Cart_controller.php]: handle in model

File: controllers/Cart/Cart_controller.php (hunk 4 above)
Comment (namdang29) [inline controllers/Cart/Cart_controller.php]: remove

File: controllers/Checkout/Checkout.controller.php [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    $password = $_POST["password"];
+
+    
+    $sql = "INSERT INTO users (username, firstName, lastName, email, phone, password) VALUES ('$username', '$firstName', '$lastName', '$email', '$phone', '$password')";
+
+    if ($conn->query($sql) === TRUE) {
+        echo "Thông tin đã được lưu vào cơ sở dữ liệu.";
+    } else {
+        echo "Lỗi: " . $sql . "<br>" . $conn->error;
+    }
```
Comment (namdang29) [inline controllers/Checkout/Checkout.controller.php]: Handle in model

File: controllers/SignUp/SignUp_controller.php [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+$dsn = "mysql:host=$hostname;dbname=$database;charset=utf8mb4";
+
+
+try {
+    $connection = new PDO($dsn, $username, $password);
+    $connection->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+} catch (PDOException $e) {
+    die("Connection failed: " . $e->getMessage());
+}
```
Comment (namdang29) [inline controllers/SignUp/SignUp_controller.php]: All of this is already in the config, no need to rewrite it

File: controllers/SignUp/SignUp_controller.php [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+    $stmt->bindParam(6, $password);
+
+
+    try {
+        $stmt->execute();
+        echo "Đăng ký thành công!";
+    } catch (PDOException $e) {
+        echo "Đã xảy ra lỗi khi đăng ký. Vui lòng thử lại sau.";
+   
+    }
```
Comment (namdang29) [inline controllers/SignUp/SignUp_controller.php]: Handle in model

File: views/SignUp/SignUp.css [hunk 11]
```diff
@@ -0,0 +1,78 @@
+body {
+    font-family: Arial, sans-serif;
+    background-color: #f5f5f5;
+    padding: 20px;
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+  }
+  h1 {
```
Comment (namdang29) [inline views/SignUp/SignUp.css:10]: Add one line between each block

File: views/SignUp/SignUp.css [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+ 
+  input[type="submit"] {
+    background-color:#e64c47;
+    color: #fff;
+    padding: 10px 20px;
+    border: none;
+    border-radius: 3px;
+    cursor: pointer;
+    width:100%;
+
```
Comment (namdang29) [inline views/SignUp/SignUp.css:55]: remove

File: controllers/Cart/Cart_controller.php (hunk 1 above)
Comment (Leho123) [reply controllers/Cart/Cart_controller.php]: dạ anh

**Pull request merged**